core/node/genesis/src/lib.rs

```diff
@@ -264,6 +264,55 @@ pub async fn is_genesis_needed(storage: &mut Connection<'_, Core>) -> Result<boo
     Ok(storage.blocks_dal().is_genesis_needed().await?)
 }
 
+pub async fn validate_genesis_params(
+    genesis_params: &GenesisParams,
+    query_client: &dyn EthInterface,
+    diamond_proxy_address: Address,
+) -> anyhow::Result<()> {
+    let hyperchain_abi = hyperchain_contract();
+    let verifier_abi = verifier_contract();
+
+    let packed_protocol_version: U256 = CallFunctionArgs::new("getProtocolVersion", ())
+        .for_contract(diamond_proxy_address, &hyperchain_abi)
+        .call(query_client)
+        .await?;
+
+    let (minor_version, patch_version) = packed_protocol_version.div_mod((1u64 << 32).into());
```

Comment:
Is there no dedicated function for that? If we will ever have to use `major`, catching all such places would be troublesome.